src/common/messageTypes.ts

```diff
@@ -0,0 +1,193 @@
+import { DiscordSnowflake } from '@sapphire/snowflake';
+import { PermissionFlagsBits, Routes, Message as DiscordJsMessage } from 'discord.js';
+import { bot } from '../bot/api';
+import { getServerChannelSettings, getServerSettings, getUserSettings, getUserServerSettings, getUserServerChannelSettings } from '../database/api';
+import { error } from '../logger';
+import { EffectiveServerChannelSettings } from './settingsTypes';
+
+// 14 days (discord's bulk deletion age threshold) (-600 sec for time to delete buffer)
+const BULK_DELETION_MAX_AGE_MILLIS: number = 1000 * (60 * 60 * 24 * 14 - 600);
+
+// =-=-=----------------=-=-=
+// .｡.:☆ message types ☆:.｡.
+// =-=-=----------------=-=-=
+
+/**
+ * Data that is compatible with the `message_ids` database table
+ */
+export interface MessageIdsData {
+  serverId: string;
+  channelId: string;
+  messageId: string;
+  authorId: string;
+}
+
+export class Message {
+  private createdAt: Date;
+  private pinned: boolean;
+
+  public static fromDiscordJsMessage(this: void, message: DiscordJsMessage): Message {
+    if (!message.guildId) {
+      throw new Error('Cannot create a Message object from a DM');
+    }
+    const msg = new Message(message.guildId, message.channelId, message.id, message.author.id);
+    if (message.pinned) {
+      msg.setPinned();
+    }
+    return msg;
+  }
+
+  public static fromMessageData(data: MessageIdsData): Message {
+    return new Message(data.serverId, data.channelId, data.messageId, data.authorId);
```

Comment:
i should add the pin check here.
for ex:
- cache pins somewhere in a set on startup & on pin update events(?) (is that a thing?)
- upon creating a new msg object, check if the pin cache set has the message id
- setPinned if so